### DISCUSSION TITLE
How to add additional info to Fulfillment card at the order details page

### DISCUSSION THREAD

**Discussion by eduarddotgg** [vendurehq/vendure#1644] [Q&A]
I have added some new fulfillment handler with some third party fulfillment service which provides additional information that I would like to show on fulfillment card. I have added custom fields for now but this solution isn't the best from UX point of ui. Is there any way to adjust fulfillment card?

**Comment by michaelbromley** [comment]
Hi! 
Did you try adding to the `args` object of the custom [fulfillment handler](https://www.vendure.io/docs/typescript-api/fulfillment/fulfillment-handler/)? This should allow you to specify data fields which can be filled out from the fulfillment dialog.

**Comment by eduarddotgg** [comment]
I would like to add here some extra info. But instead of form elements I would like to add text components or something like that
<img width="373" alt="Screen Shot 2022-07-04 at 10 24 32" src="https://user-images.githubusercontent.com/4650029/177114137-5e9286cb-defd-4092-b36a-5bbad2bd610a.png">

**Reply by michaelbromley** [reply]
Ah ok. I think you should be able to use the `ui` property of you custom field config to specify custom components to display your custom field data. You can read more about that here: https://www.vendure.io/docs/developer-guide/customizing-models/#customfield-ui-form-inputs

**Reply by eduarddotgg** [reply]
Thanks!
